push-notifications.js

```diff
@@ -106,8 +134,90 @@ PushNotifications.prototype.publish = function(interests, publishRequest) {
     return doRequest(payload, options);
 };
 
+PushNotifications.prototype.publishToUsers = function(users, publishRequest) {
+    if (users === undefined || users === null) {
+        return Promise.reject(new Error('users argument is required'));
+    }
+    if (!(users instanceof Array)) {
+        return Promise.reject(new Error('users argument is must be an array'));
+    }
+    if (users.length < 1) {
+        return Promise.reject(
+            new Error(
+                'Publish requests must target at least one interest to be delivered'
+            )
+        );
+    }
+    if (users.length > USERS_ARRAY_MAX_LENGTH) {
+        return Promise.reject(
+            new Error(
+                `Number of users (${
+                    users.length
+                }) exceeds maximum of ${USERS_ARRAY_MAX_LENGTH}.`
+            )
+        );
+    }
+    if (publishRequest === undefined || publishRequest === null) {
+        return Promise.reject(new Error('publishRequest argument is required'));
+    }
+    for (const user of users) {
+        if (typeof user !== 'string') {
+            return Promise.reject(new Error(`user ${user} is not a string`));
+        }
+        if (user.length > INTEREST_STRING_MAX_LENGTH) {
+            return Promise.reject(
+                new Error(
+                    `user ${user} is longer than the maximum of ` +
+                        `${INTEREST_STRING_MAX_LENGTH} characters`
+                )
+            );
+        }
+    }
+
+    publishRequest.users = users;
+    const payload = JSON.stringify(publishRequest);
+    const options = {
+        host: this.endpoint,
+        path: `/publish_api/v1/instances/${this.instanceId}/users`,
+        port: 443,
```

Comment:
Now that this port stuff is done in multiple places it could be moved to doRequest